File: matlab/ECE345/HW01/p2.m
% Assignment 1-2
% In rolling a die four times, what is the
% odds that at least one 6 would appear? 

NUM_TRIALS = 1000000;

success_cnt = 0;
total_cnt = 0;

for i = 1:NUM_TRIALS
    for j = 1:4
        roll = ceil(rand() * 6);

        if roll == 6
            success_cnt++;
            break
        end
    end

    total_cnt++;
end

disp(strcat('SUCCESS/TOTAL=', num2str(success_cnt), '/', num2str(total_cnt)))
disp(strcat(num2str(100 * success_cnt / total_cnt), '%'))

File: matlab/ECE345/HW01/p2.c
#include <stdio.h>
int main(int argc, char const* argv[]) {
    srand(time(NULL));

    int NUM_TRIALS = 1000000;

    int success_cnt = 0;
    int total_cnt = 0;

    int i;
    for (i = 0; i < NUM_TRIALS; i++) {
        int j;
        for (j = 0; j < 4; j++) {
            int roll = (int)(rand() % 6) + 1;

            if (roll == 6) {
                success_cnt++;
                break;
            }
        }

        total_cnt++;
    }

    printf("SUCCESS/TOTAL=%d/%d\n", success_cnt, total_cnt);
    printf("%f%%\n", 100.0 * success_cnt / total_cnt);
    return 0;
}
